internal/hcp/internal/controllers/link/finalizer.go

```diff
@@ -8,28 +8,29 @@ import (
 	"os"
 	"path/filepath"
 
+	"github.com/hashicorp/consul/agent/hcp"
 	"github.com/hashicorp/consul/agent/hcp/bootstrap"
 	"github.com/hashicorp/consul/internal/controller"
 	"github.com/hashicorp/consul/internal/resource"
 	"github.com/hashicorp/consul/proto-public/pbresource"
 )
 
-func cleanup(ctx context.Context, rt controller.Runtime, res *pbresource.Resource, dataDir string) error {
+func cleanup(rt controller.Runtime, hcpManager hcp.Manager, dataDir string) error {
 	rt.Logger.Trace("cleaning up link resource")
+
+	rt.Logger.Debug("stopping HCP manager")
+	hcpManager.Stop()
+
 	if dataDir != "" {
 		hcpConfigDir := filepath.Join(dataDir, bootstrap.SubDir)
 		rt.Logger.Debug("deleting hcp-config dir", "dir", hcpConfigDir)
 		err := os.RemoveAll(hcpConfigDir)
 		if err != nil {
+			rt.Logger.Error("failed to delete hcp-config dir", "dir", hcpConfigDir, "err", err)
 			return err
 		}
 	}
 
-	err := ensureDeleted(ctx, rt, res)
-	if err != nil {
-		return err
-	}
-
```

Comment:
Why'd we delete this? I thought the Link would never actually be deleted unless we remove the finalizer, then call Delete... but I must be missing something because I see the tests are passing.

Reply:
disregard, I see that you added this above.